src/core/util/tools.py

```
@@ -23,21 +25,41 @@
-        Constants.log.info(f"[shell] {line}")
-        info += line
-
-        if line == "" or subprocess.Popen.poll(cmd) == 0:  # 判断子进程是否结束
-            break
-
-    return info
+def run_py_file(payload: str, cwd: str, log_ignore_regex: str | None = None) -> str:
+    Constants.log.info(f'[shell] cd "{cwd}"')
+    Constants.log.info(f'[shell] python -X utf8 {payload}')
+
+    args = [sys.executable, '-X', 'utf8'] + shlex.split(payload)
```

Ruff (0.13.1)

Consider iterable unpacking instead of concatenation

Replace with iterable unpacking

(RUF005)